### DISCUSSION TITLE
大家还希望 PanelSwitchHelper 新增哪些功能？欢迎大家盖楼。

### DISCUSSION THREAD

**Discussion by YummyLau** [DSAppTeam/PanelSwitchHelper#120] [General]
欢迎大家提出任何建设性意见及建议，一起讨论。

**Comment by lcuwx2016** [comment]
首先感谢楼主的努力！有一个小建议，目前打开面板后先打开输入法，再关闭输入法的时候界面还是会闪现面板内容的，这个面板内容的闪现有可能隐藏吗？希望得到楼主回复~

**Reply by YummyLau** [reply]
@lcuwx2016 你指的是输入法在隐藏的时候，能看到背后的面板内容么

**Reply by lcuwx2016** [reply]
对，输入法隐藏的过程中，可以看到背后的面板一闪而过

**Reply by YummyLau** [reply]
其实这部分内容是跟在内容下面，输入法隐藏时，由于界面下滑的速度比输入法慢，导致你会看到。是可以处理的。

**Reply by lcuwx2016** [reply]
那这能划到开发计划里吗~

**Reply by YummyLau** [reply]
下个版本可处理下。这一两周可以支持

**Reply by lcuwx2016** [reply]
辛苦了哈

**Reply by wb-hwang** [reply]
现在还会，请问这个是加了处理了吗？

**Comment by originlan** [comment]
使用DialogFragment，点击外部setCanceledOnTouchOutside只能隐藏PanelContainer，希望能够跟情况隐藏PanelContainer和LinearContentContainer，因为现在setCanceledOnTouchOutside = true 无效

**Reply by YummyLau** [reply]
目前 PanelContainer 是可以根据状态隐藏的，具体是哪些场景。 希望进一步了解一下。

**Comment by zkoalas** [comment]
楼主能增加一个小功能吗?在聊天界面中,当输入法弹出时,第一次输入的聊天内容,看不见,我细心观察微信功能,第一次聊天,是可以看见的,在demo中adapter的count=50,修改成1或者0的时候进行测试,会看到这种现象,-》第一次聊天内容看不到 -〉 解决方案 -》 contentContainer 减去键盘高度, 当然某些细节也需要调整一下,想问一下楼主,这个功能,能增加吗?  希望得到楼主回复~

**Reply by YummyLau** [reply]
其实这个问题已经有api支持的，可以看下 addContentScrollMeasurer api

**Comment by davidgerka** [comment]
请问大神可以增加多个EditText输入表情的功能吗，应用场景是，App要发表文章时，需要在界面上输入文章的标题、文章的内容，这里就会有两个EditText都需要输入表情了。现在是直接传EditText实例进去表情键盘的，弄一个接口出来实现？

**Comment by zengcanxiang** [comment]
在定义的panel能够直接知道当前被隐藏或者显示出来么。否则需要再监听里面，通过固定的id获取panel，在手动调用方法。这样，存在耦合。而且，监听里面各种情况，存在比较多的重复代码

**Reply by YummyLau** [reply]
可以直接获取状态的

**Comment by FZ985** [comment]
面板可以自定义高度吗，比如类似微信的表情面板，比键盘高度高一点

**Reply by FZ985** [reply]
我扩展了 PanelHeightMeasurer、PanelHeightMeasurerBuilder两个类添加了forceUseTargetPanelDefaultHeight外部回调，然后修改了getCompatPanelHeight函数好像能达到微信的不同面板不同高度的效果实现

**Reply by HeroZ-Dodge** [reply]
* 可选，可不设置
                     * 面板默认高度设置,输入法显示后会采纳输入法高度为面板高度，否则则以框架内部默认值为主
                     */
                    .addPanelHeightMeasurer(new PanelHeightMeasurer() {

                        /**
                         * false 为不同步输入法高度
                         * @return
                         */
                        @Override
                        public boolean synchronizeKeyboardHeight() {
                            return false;
                        }

                        @Override
                        public int getTargetPanelDefaultHeight() {
                            return DisplayUtils.dip2px(ChatSuperActivity.this, 500f);
                        }

                        @Override
                        public int getPanelTriggerId() {
                            return R.id.add_btn;
                        }
                    })

PanelHeightMeasurer 将 synchronizeKeyboardHeight方法返回 false 时，表示不同步键盘高度，而是使用 getTargetPanelDefaultHeight 方法返回的高度作为面板高度

可以参考 Demo 中 ChatSuperActivity 这个类的使用场景

**Comment by FZ985** [comment]
表情面板是否能实现向上拖拽显示更多，类似微信、抖音
